crates/matrix-sdk/tests/integration/encryption/verification.rs

```diff
@@ -202,3 +346,152 @@ async fn test_own_verification() {
+    {
+        let alice_olm = alice.olm_machine_for_testing().await;
+        let alice_olm = alice_olm.as_ref().unwrap();
+        alice_olm.update_tracked_users([bob_user_id.as_ref()]).await.unwrap();
+    }
+
+    // Run a sync so we do send outgoing requests, including the /keys/query for
+    // getting bob's identity.
+    let mut sync_response_builder = SyncResponseBuilder::new();
+    mock_sync(&server.server, sync_response_builder.build_json_sync_response(), None).await;
+    alice.sync_once(Default::default()).await.unwrap();
+
+    // From the point of view of Alice, Bob now has a device.
+    let alice_bob_device = alice
+        .encryption()
+        .get_device(&bob_user_id, &bob_device_id)
+        .await
+        .unwrap()
+        .expect("alice sees bob's device");
+
+    assert!(!alice_bob_device.is_verified());
+    assert!(!alice_bob_device.is_deleted());
+    assert!(alice_bob_device.verify().await.is_err(), "can't sign the device of another user");
+
+    let alice_bob_ident = alice
+        .encryption()
+        .get_user_identity(&bob_user_id)
+        .await
+        .unwrap()
+        .expect("alice sees bob's identity");
+
+    alice_bob_ident.verify().await.unwrap();
+
+    // Notify Alice's devices that some identify changed, so it does another
+    // /keys/query request.
+    {
+        let alice_olm = alice.olm_machine_for_testing().await;
+        let alice_olm = alice_olm.as_ref().unwrap();
+        let changed_devices = &assign!(DeviceLists::default(), {
+            changed: vec![bob_user_id.clone()]
+        });
+        alice_olm
+            .receive_sync_changes(EncryptionSyncChanges {
+                to_device_events: Default::default(),
+                changed_devices,
+                one_time_keys_counts: &Default::default(),
+                unused_fallback_keys: Default::default(),
+                next_batch_token: None,
+            })
+            .await
+            .unwrap();
+    }
+
+    mock_sync(&server.server, sync_response_builder.build_json_sync_response(), None).await;
+    alice.sync_once(Default::default()).await.unwrap();
+
+    let alice_bob_ident = alice
+        .encryption()
+        .get_user_identity(&bob_user_id)
+        .await
+        .unwrap()
+        .expect("alice sees bob's identity");
+    assert_eq!(alice_bob_ident.user_id(), bob_user_id);
+    assert!(alice_bob_ident.is_verified());
+
+    let master_pub_key = alice_bob_ident.master_key();
+    assert_eq!(master_pub_key.user_id(), bob_user_id);
+    assert!(!master_pub_key.keys().is_empty());
+    assert_eq!(master_pub_key.keys().iter().count(), 1);
+
+    let alice_bob_device = alice
+        .encryption()
+        .get_device(&bob_user_id, &bob_device_id)
+        .await
+        .unwrap()
+        .expect("alice sees bob's device");
+    assert!(alice_bob_device.is_verified()); // fails?
+    assert!(alice_bob_device.is_verified_with_cross_signing()); // fails?
```

Comment:
*narrator's voice*: they don't fail anymore!

```suggestion
    assert!(alice_bob_device.is_verified());
    assert!(alice_bob_device.is_verified_with_cross_signing());
```